tests/cluster_tests/pbsdsh_custom/pbsdsh_mode.py

```diff
@@ -15,55 +15,93 @@
 
 import os
 import Simulation
+import string
+
+def createAndRunQSUB(runInfoDict):
+  """
+    Generates a PBS qsub command to run the simulation
+    @ In, runInfoDict, dict, dictionary of run info.
+    @ Out, remoteRunCommand, dict, dictionary of command.
+  """
+  # Check if the simulation has been run in PBS mode and, in case, construct the proper command
+  #while true, this is not the number that we want to select
+  coresNeeded = runInfoDict['batchSize']*runInfoDict['NumMPI']
+  #batchSize = runInfoDict['batchSize']
+  frameworkDir = runInfoDict["FrameworkDir"]
+  ncpus = runInfoDict['NumThreads']
+  jobName = runInfoDict['JobName'] if 'JobName' in runInfoDict.keys() else 'raven_qsub'
+  #check invalid characters
+  validChars = set(string.ascii_letters).union(set(string.digits)).union(set('-_'))
+  if any(char not in validChars for char in jobName):
+    raise IOError('JobName can only contain alphanumeric and "_", "-" characters! Received'+jobName)
+  #check jobName for length
+  if len(jobName) > 15:
+    jobName = jobName[:10]+'-'+jobName[-4:]
+    print('JobName is limited to 15 characters; truncating to '+jobName)
```

Comment:
Technically, we now pass in the messageHander to the init function, so it would be possible to use the messageHandler instead.  I can change it if you want, but these are outside of the core portion of RAVEN.